File: design-patterns/ts/strategy/transport/bike.ts
// bike.ts

import Transport from "./transport";

class Bike implements Transport {
    start(): void {
        console.log("Bike started");
    }
    
    stop(): void {
        console.log("Bike stopped");
    }

    repair(): void {
        console.log("Bike repair");
    }

    getInfo(): void {
        console.log("Transport type: Bike");
        console.log("Number of wheels: 2");
        console.log("Average Weight: 700 Pounds");
    }

    operate(): void {
        console.log("Riding Bike ............");
    }
}

export default Bike;

File: design-patterns/go/strategy/transport/bike.go
// bike.go

package main

import "fmt"

type Bike struct {
}

func NewBike() (bike *Bike) {
	bike = &Bike{}
	return
}

func (bike *Bike) GetInfo() {
	fmt.Println("Transport type: Bike")
	fmt.Println("Number of wheels: 2")
	fmt.Println("Average Weight: 700 Pounds")
}

func (bike *Bike) Operate() {
	fmt.Println("Riding Bike ............")
}

func (bike *Bike) Repair() {
	fmt.Println("Bike repair")
}

func (bike *Bike) Start() {
	fmt.Println("Bike started")
}

func (bike *Bike) Stop() {
	fmt.Println("Bike stopped")
}
